src/avt_camera/scripts/stereo_camera.py

```diff
@@ -0,0 +1,147 @@
+		self._left_info_pub = rospy.Publisher(rospy.get_name()+"/left" + "/" + self.INFO_TOPIC_NAME, CameraInfo, queue_size=10)
+		self._right_pub = rospy.Publisher(rospy.get_name()+"/right/" + self.RAW_TOPIC_NAME, Image, queue_size=10)
+		self._right_info_pub = rospy.Publisher(rospy.get_name()+"/right" + "/" + self.INFO_TOPIC_NAME, CameraInfo, queue_size=10)
+		self._bridge = CvBridge()
+        #self._camera_id = rospy.get_param('~camera_id', None)
+		self._left_camera_id = "DEV_000F315C1ED5"
+		self._right_camera_id = "DEV_000F315C1ED8"
+
+	def run(self):
+        
+		with Vimba() as vimba:
+			system = vimba.getSystem()
+			system.runFeatureCommand("GeVDiscoveryAllOnce")
+			rospy.sleep(0.1)
+
+			camera_ids = vimba.getCameraIds()
+			if not camera_ids:
+				rospy.logerr("Sorry: Cameras were not found.")
+				sys.exit(0)
+
+			for cam_id in camera_ids:
+				rospy.loginfo("Camera found: " + cam_id)
+		
+
+			if self._left_camera_id is None :
+				self._left_camera_id = camera_ids[0]
+			elif self._left_camera_id not in camera_ids:
+				rospy.logerr("Requested camera ID (" + self._left_camera_id + ") not found, sorry")
+				sys.exit(0)
+			self._left_info_manager = CameraInfoManager(cname=self._left_camera_id, namespace = "left", url="package://avt_camera/calibrations/${NAME}.yaml")
+			self._left_info_manager.loadCameraInfo()
+	    
+			if self._right_camera_id is None :
+				self._right_camera_id = camera_ids[0]
+			elif self._right_camera_id not in camera_ids:
+				rospy.logerr("Requested camera ID (" + self._right_camera_id + ") not found, sorry")
+				sys.exit(0)
+			self._right_info_manager = CameraInfoManager(cname=self._right_camera_id, namespace = "right", url="package://avt_camera/calibrations/${NAME}.yaml")
+			self._right_info_manager.loadCameraInfo()
+			c0 = vimba.getCamera(self._left_camera_id)
+			c0.openCamera()
+			c1 = vimba.getCamera(self._right_camera_id)
+			c1.openCamera()
+
+			try:
+                #gigE camera
+				rospy.loginfo("Packet Size: " + str(c0.GevSCPSPacketSize))
+				rospy.loginfo("Stream Bytes Per Second: " + str(c0.StreamBytesPerSecond))
+				c0.runFeatureCommand("GVSPAdjustPacketSize")
+				c0.StreamBytesPerSecond = 100000000
+				c1.runFeatureCommand("GVSPAdjustPacketSize")
+				c1.StreamBytesPerSecond = 100000000
+			except:
+                #not a gigE camera
+				pass
+
+            #set pixel format
+			c0.PixelFormat="Mono8"
+			c0.AcquisitionMode = "Continuous"
+			c0.ExposureAuto = "Continuous"
+			c0.Width = 1210
+			c0.Height = 760
+			lframe = c0.getFrame()
+			lframe.announceFrame()
+			
+			#set left (c1)
+			c1.PixelFormat="Mono8"
+			c1.AcquisitionMode="Continuous"
+			c1.ExposureAuto="Continuous"
+			c1.Width = 1210
+			c1.Height = 760
+			rframe = c1.getFrame()
+			rframe.announceFrame()
+
+			c0.startCapture()
+			c1.startCapture()
+
+			framecount = 0
+			droppedframes = []
+            
+			c0.runFeatureCommand("AcquisitionStart") 
+			c1.runFeatureCommand("AcquisitionStart")
```

Comment:
there's a lot of code duplication here, consider using a function or a loop to do the setup on both cameras simultaneously

Reply:
Most, if not all duplication was removed in 3bec872. Last bit (between monocamera and synchronouscamera) can be figured out later, if we decide that synchronous camera is good enough without mono camera.